src/MSALWrapper/Locked.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+
+    /// <summary>
+    /// A method for executing a <see cref="Func{T}"/> under an inter-process lock.
+    /// </summary>
+    public static class Locked

```

Comment:
Just curious why did we name it `Locked` instead of `Lock`?

Reply:
Partially English preference. I usually find myself saying "we want to do this in a locked fashion", and partially to be different from `Lock` because `lock` (lowercase) is a keyword.